src/callbacks_step/analysis.jl

```diff
@@ -336,7 +336,7 @@ end
 # the derivative of the velocity `v_x`.
 function analyze(quantity::Union{typeof(energy_total_modified), typeof(entropy_modified)},
                  q, t, semi::Semidiscretization)
-    integrate_quantity(quantity, q, semi)
+    integrate_quantity!(semi.cache.tmp1, quantity, q, semi)
```

Comment:
We mutate `semi.cache.tmp1` here, but `analyze` does not have a `!`, which is necessary to make the dispatch work. Should we leave it like this? We could also call both versions `analyze!` eventhough the other one is non-mutating. Any thoughts, @ranocha?

Reply:
Now that I think of it: We also allocate a vector for the other cases (other `quantity` than `energy_total_modified`) by calling `integrate_quantity`. We could also call `integrate_quantity!` in those cases and pass `semi.cache.tmp1` as first argument. In that case `analyze` would always mutate and the problem above would be solved.

Reply:
I think we could even completely get rid of the dispatching in this file by defining
```julia
function analyze!(semi::Semidiscretization, quantity, q, t)
  integrate_quantity!(semi.cache.tmp1, quantity, q, semi)
end
```
and slightly rewriting the ` integrate_quantity!` function for the usual `quantities`, which are not `energy_total_modified` or `entropy_modified`.

Reply:
I implemented that in 8fec4157ee65ffe67eb8e1a483184de6c4527b1e. This also simplifies the relaxation callback, which is quite nice.